src/internal/modules/invoice-accounts/lib/helpers.js

```diff
@@ -11,11 +13,15 @@ const getFormTitleCaption = (licenceNumber) => {
 const processCompanyFormData = (request, regionId, companyId, formData) => {
   const { selectedCompany, companySearch } = forms.getValues(formData);
   if (selectedCompany === 'company_search') {
-    // TODO place holder -- route does not exist
-    return `company-search?filter=${companySearch}`;
+    return `contact-search?filter=${companySearch}`;
   } else {
     const agentId = selectedCompany === companyId ? null : selectedCompany;
-    dataService.sessionManager(request, regionId, companyId, { agent: agentId });
+    if (agentId) {
```

Comment:
I think the above line was fine as it is. You only need to set the agent either to null or if the selected company id is not the same as the main company id then set the agent = to the selected company id.

Reply:
> I think the above line was fine as it is.

The purpose of the change is to: 
1. inject the name of the agent company into the viewData object, so that the agent company name appears in the UI under `Select an address for COMPANY NAME`
2. prevent the company from being entered as its own agent

> on line 21 below we don't change the companyId because this is the original and main company for the licence.

Ooo. I think I disagree: I think the agent object should only represent the agent (if there is one). The original company ID can always be derived from the request parameters, so there's no need to put the original companyId in the session.agent object.